File: src/main/scala/emasher/sockets/client/ClientProxy.java
package emasher.sockets.client;

import cpw.mods.fml.client.registry.ClientRegistry;
import cpw.mods.fml.relauncher.Side;
import emasher.sockets.CommonProxy;
import emasher.sockets.SocketsMod;
import emasher.sockets.TileSocket;
import emasher.sockets.TileTempRS;
import emasher.sockets.packethandling.*;
import emasher.sockets.pipes.TilePipeBase;

public class ClientProxy extends CommonProxy {
	@Override
	public void registerRenderers() {
		ClientRegistry.bindTileEntitySpecialRenderer( TileSocket.class, SocketRenderer.instance );
		ClientRegistry.bindTileEntitySpecialRenderer( TilePipeBase.class, PipeRenderer.instance );
		ClientRegistry.bindTileEntitySpecialRenderer( TileTempRS.class, TempRSRenderer.instance() );
	}

	@Override
	public void registerMessages() {
		SocketsMod.network.registerMessage( RequestInfoFromServerMessage.Handler.class, RequestInfoFromServerMessage.class, 6, Side.SERVER );

		SocketsMod.network.registerMessage( SocketStateMessage.Handler.class, SocketStateMessage.class, 0, Side.CLIENT );
		SocketsMod.network.registerMessage( SocketItemMessage.Handler.class, SocketItemMessage.class, 1, Side.CLIENT );
		SocketsMod.network.registerMessage( SocketFluidMessage.Handler.class, SocketFluidMessage.class, 2, Side.CLIENT );
		SocketsMod.network.registerMessage( PipeColourMessage.Handler.class, PipeColourMessage.class, 3, Side.CLIENT );
		SocketsMod.network.registerMessage( AdapterSideMessage.Handler.class, AdapterSideMessage.class, 4, Side.CLIENT );
		SocketsMod.network.registerMessage( ChangerSideMessage.Handler.class, ChangerSideMessage.class, 5, Side.CLIENT );
	}
}

File: src/main/scala/emasher/client/ClientProxy.scala
package emasher.client

import cpw.mods.fml.client.registry.{ClientRegistry, RenderingRegistry}
import cpw.mods.fml.common.Loader
import cpw.mods.fml.relauncher.Side
import emasher.nei.NEIHandler
import emasher.{EngineersToolbox, CommonProxy}
import emasher.entities.EntitySmokeBomb
import emasher.packethandling._
import emasher.tileentities.{TileTempRS, TilePipeBase, TileSocket}
import net.minecraft.client.renderer.entity.RenderSnowball

class ClientProxy extends CommonProxy {
  override def registerRenderers(): Unit = {
    RenderingRegistry.registerEntityRenderingHandler( classOf[EntitySmokeBomb], new RenderSnowball( emasher.items.Items.smokeGrenade ) )
    ClientRegistry.bindTileEntitySpecialRenderer( classOf[TileSocket], SocketRenderer.instance )
    ClientRegistry.bindTileEntitySpecialRenderer( classOf[TilePipeBase], PipeRenderer.instance )
    ClientRegistry.bindTileEntitySpecialRenderer( classOf[TileTempRS], TempRSRenderer.instance )
  }

  override def registerMessages(): Unit = {
    EngineersToolbox.network.registerMessage( classOf[RequestInfoFromServerMessage.Handler], classOf[RequestInfoFromServerMessage], 6, Side.SERVER )

    EngineersToolbox.network.registerMessage( classOf[SocketStateMessage.Handler], classOf[SocketStateMessage], 0, Side.CLIENT )
    EngineersToolbox.network.registerMessage( classOf[SocketItemMessage.Handler], classOf[SocketItemMessage], 1, Side.CLIENT )
    EngineersToolbox.network.registerMessage( classOf[SocketFluidMessage.Handler], classOf[SocketFluidMessage], 2, Side.CLIENT )
    EngineersToolbox.network.registerMessage( classOf[PipeColourMessage.Handler], classOf[PipeColourMessage], 3, Side.CLIENT )
    EngineersToolbox.network.registerMessage( classOf[AdapterSideMessage.Handler], classOf[AdapterSideMessage], 4, Side.CLIENT )
    EngineersToolbox.network.registerMessage( classOf[ChangerSideMessage.Handler], classOf[ChangerSideMessage], 5, Side.CLIENT )
  }

  override def registerNEI(): Unit = {
    if (Loader.isModLoaded("NotEnoughItems")) {
      val handler: NEIHandler = new NEIHandler
      handler.loadConfig()
    }
  }
}
